packages/suite/src/views/wallet/sign-verify/index.tsx

```diff
@@ -1,207 +1,170 @@
-                            rows={4}
-                            maxRows={4}
-                            maxLength={255}
-                        />
+                        {page === 'sign' ? (
+                            <SignAddressInput
+                                name="path"
+                                label={<Translation id="TR_ADDRESS" />}
+                                account={selectedAccount.account}
+                                revealedAddresses={revealedAddresses}
+                                error={formErrors.path}
+                                {...pathField}
+                            />
+                        ) : (
+                            <Input
+                                name="address"
+                                label={<Translation id="TR_ADDRESS" />}
+                                type="text"
+                                state={errorState(formErrors.address)}
+                                bottomText={formErrors.address}
+                                {...addressField}
+                            />
+                        )}
                     </Row>
                     <Row>
                         <Textarea
+                            name="signature"
                             label={<Translation id="TR_SIGNATURE" />}
-                            name="verifySignature"
-                            value={signVerify.verifySignature}
-                            onChange={handleInputChange}
+                            maxLength={MAX_LENGTH_SIGNATURE}
+                            innerRef={signatureRef}
+                            readOnly={page === 'sign'}
+                            state={errorState(formErrors.signature)}
+                            bottomText={formErrors.signature}
                             rows={4}
                             maxRows={4}
-                            maxLength={255}
                         />
                     </Row>
-                    <RowButtons>
-                        <StyledButton onClick={clearVerify}>
-                            <Translation id="TR_CLEAR" />
-                        </StyledButton>
-                        <StyledButton
-                            // isDisabled={!!verifyAddressError || !device.connected}
-                            isDisabled
-                            onClick={() => {
-                                if (signVerify.errors.length <= 0) {
-                                    verify(
-                                        signVerify.verifyAddress,
-                                        signVerify.verifyMessage,
-                                        signVerify.verifySignature,
-                                    );
-                                }
-                            }}
-                        >
-                            <Translation id="TR_VERIFY" />
+                    <Row>
+                        <StyledButton type="submit">
```

Comment:
not a big deal but I believe that this button should be disabled if there is still some kind of communication with device in progress. Possible edge case: 
1. enable all coins on all all seed (or any other larger account)
2. reload page 
3. quickly try to sign some message
4. get error "device call in progress"

I think there is `useDevice` hook which may help with this.

Reply:
live example in #4300

Reply:
Agreed. Buton disabling implemented, e2e tests merged and extended. Everything seems to be working.